File: VideoCallSDK/app/src/main/java/com/vnptit/videocallsample/db/ContactDao.java
package com.vnptit.videocallsample.db;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.vnptit.videocallsample.model.Contact;

import java.util.List;

@Dao
public interface ContactDao {
    @Query("SELECT * FROM Contact")
    LiveData<List<Contact>> getAll();

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    void insert(Contact contact);

    @Update
    void update(Contact contact);

    @Delete
    void delete(Contact contact);
}

File: VideoCallSDK/kotlin/src/main/java/com/vnptit/videocallsample/db/ContactDao.kt
package com.vnptit.videocallsample.db

import androidx.lifecycle.LiveData
import androidx.room.*
import com.vnptit.videocallsample.model.ContactCall

/**
 * Created by Ông Hoàng Nhật Phương on 1/26/2021.
 * VNPT COM
 * phuonghn@vnpt.vn
 */
@Dao
interface ContactDao {

    @Query("SELECT * FROM Contact")
    fun getAll(): LiveData<List<ContactCall>>

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    fun insertContact(contactCall: ContactCall)

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    fun insertAll(contactCall: List<ContactCall>)

    @Update
    fun update(contactCall: ContactCall)

    @Delete
    fun delete(contactCall: ContactCall)
}